neo/app/controllers/Application.scala

```diff
@@ -83,10 +86,87 @@ class Application @Inject()(val wsClient: WSClient,
     }
   }
 
+  def getCity(NELat: Double, SWLat: Double, NELng: Double, SWLng: Double) = Action{
+    Ok(Application._getCity(NELat, SWLat, NELng, SWLng, cities))
+  }
+
   class Listener extends Actor {
     def receive = {
       case d: DeadLetter => println(d)
     }
   }
 
 }
+
+object Application{
+  def loadCity(file: File): List[JsValue] = {
+    val stream = new FileInputStream(file)
+    val json = Json.parse(stream)
+    val features = "features"
+    val geometry = "geometry"
+    val type_str = "type"
+    val coordinates = "coordinates"
+    val Polygon = "Polygon"
+    val MultiPolyton = "MultiPolygon"
+    val values = (json \ features).as[List[JsObject]]
+    val newValues = List.newBuilder[JsValue]
+    for(n <- values.indices) {
```

Comment:
using `indices` introduce the extra overhead of looking up element by id.
change to `for (value <- values)` will be good.

Furthermore, the `newValues` and the `values` is just a mapping relation. Instead of creating a `ListBuilder` you can write
```
val newValues = values.map { value =>
 values.geotype match {
    case Polygon => ...
    case MultiPolygon => ...
    case _ => throw exception ...
  }
}
```